### ISSUE TITLE
joss review: error in tests, but test "ok"

### ISSUE THREAD

**Issue by wlav** [scikit-hep/root_numpy#328]
When running the tests, I get:

root_numpy.tests.test_evaluate.test_evaluate_func` ... input_line_99:1:55: error: use of undeclared identifier 'f'
Double_t TFormula____id6180733986148604910(){ return {f} ; }
                                                      ^

And yet the result is:

ok

The 'error' may well be ignorable, but recommend to then suppress it. That said, the code really isn't valid C++, so something may be lurking underneath and thus the test should not have succeeded.

**Comment by ndawe** [comment]
Thanks @wlav for starting the JOSS review! I'll try to respond to each issue as quickly as possible, but I'll be back in Melbourne on July 27 so I might not be able to respond to all of them until after that.

This particular error is expected and is the result of the test where the function cannot be evaluated:

```python
assert_raises(ValueError, rnp.evaluate, "f", arr_1d)
```

For example:

```python
>>> import root_numpy as rnp
>>> import numpy as np
>>> rnp.evaluate("f", np.ones(10))
input_line_35:1:55: error: use of undeclared identifier 'f'
Double_t TFormula____id6180733986148604910(){ return {f} ; }
                                                      ^
Error in <TFormula::Eval>: Can't find TFormula____id6180733986148604910 function prototype with arguments 
Error in <TFormula::ProcessFormula>: "f" has not been matched in the formula expression
Error in <TFormula::ProcessFormula>: Formula "f" is invalid !
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/endw/.local/lib/python2.7/site-packages/root_numpy/_evaluate.py", line 108, in evaluate
    raise ValueError("invalid formula expression")
ValueError: invalid formula expression
```

This seems to be an error originating within Cling. Do you know of a way to silence Cling from writing to stdout/stderr?

**Comment by wlav** [comment]
I'm not aware of a public API to do that and passing the Cling redirection syntax on gInterpreter.ProcessLine() seems to have no effect. However, you can temporarily redirect stderr output to /dev/null,  Example:

```python
import os, sys
import root_numpy as rnp
import numpy as np
save = file( '/dev/null', 'w' )
os.dup2(sys.stderr.fileno(), save.fileno())
shutup = file( '/dev/null', 'w' )
print >> sys.stderr, 'loud'
os.dup2(shutup.fileno(), sys.stderr.fileno())
print >> sys.stderr, 'silent'
try:
   rnp.evaluate("f", np.ones(10))
except ValueError:
   pass
os.dup2(save.fileno(), sys.stderr.fileno())
print >> sys.stderr, 'loud'
```

Of course, this silences all errors, not just the ones you expect or only the ones from Cling, so capturing the output in an unlinked file for printing if need be, may serve better.

**Comment by ndawe** [comment]
Thanks @wlav! I've fixed this in #335
